Source/PINCacheConfigurationInternal.m

```diff
@@ -0,0 +1,63 @@
+//
+//  PINCacheConfigurationInternal.m
+//  PINCache
+//
+//  Created by Vladimir Solomenchuk on 3/19/20.
+//  Copyright © 2020 Pinterest. All rights reserved.
+//
+
+#import "PINCacheConfigurationInternal.h"
+
+@interface PINCacheConfiguration(PINCacheConfiguration)
++ (PINCacheExperimentalFeatures) defaultConfiguration;
+@end
+
+@implementation PINCacheConfiguration
++ (PINCacheExperimentalFeatures) defaultConfiguration {
+    return PINCacheExperimentalFeatureAll;
+}
+@end
+
+
+static PINCacheConfigurationManager *PINCacheSharedConfigurationManager;
+static dispatch_once_t PINCacheSharedConfigurationManagerOnceToken;
+
+NS_INLINE PINCacheConfigurationManager *PINCacheConfigurationManagerGet() {
+  dispatch_once(&PINCacheSharedConfigurationManagerOnceToken, ^{
+    PINCacheSharedConfigurationManager = [[PINCacheConfigurationManager alloc] init];
+  });
+  return PINCacheSharedConfigurationManager;
+}
+
+@implementation PINCacheConfigurationManager {
+  _Atomic(PINCacheExperimentalFeatures) _activatedExperiments;
+}
+
+- (instancetype)init
+{
+  if (self = [super init]) {
+    if ([PINCacheConfiguration respondsToSelector:@selector(cacheConfiguration)]) {
+      _activatedExperiments = [PINCacheConfiguration cacheConfiguration];
+    } else {
+      _activatedExperiments = [PINCacheConfiguration defaultConfiguration];
+    }
+  }
+  return self;
+}
+
+- (BOOL)activateExperimentalFeature:(PINCacheExperimentalFeatures)requested
```

Comment:
Is this method activating the experiment? Or indicating whether it should be active?

Reply:
this method checks is an experiment is active. In code developers should use `PINCacheHasActiveExperimentalFeature` function. e.g.
```objc
if (PINCacheHasActiveExperimentalFeature(PINCacheHasActiveExperimentalFeatureDefaultDoNotRemoveAllObjectOnMemoryWarning)) {
   _removeAllObjectsOnMemoryWarning = NO;
} else {
   _removeAllObjectsOnMemoryWarning = YES;
}
```